rust/abacus-base/src/metrics/core.rs

```diff
@@ -292,65 +263,117 @@ impl CoreMetrics {
         )?)
     }
 
-    /// Call with RPC duration after it is complete
-    pub fn rpc_complete(&self, chain: &str, method: &str, duration: Duration) {
-        self.rpc_latencies
-            .with_label_values(&[chain, method, &self.agent_name])
-            .observe(duration.as_secs_f64())
-    }
-
-    /// Gauge for measuring the last known message leaf index
+    /// Reports the current highest message leaf index at multiple phases of the
+    /// relaying process. There may be messages that have not reached a certain
+    /// stage, such as being fully processed, even if the reported index is
+    /// higher than that message's index.
+    ///
+    /// Some phases are not able to report the remote chain, but origin chain is
+    /// always reported.
+    ///
+    /// Labels:
+    /// - `phase`: The the leaf index is being tracked at, see below.
+    /// - `origin`: Origin chain the index comes from.
+    /// - `remote`: Remote chain for the message. This will skip values because
+    ///   the indexes are contiguous by origin not remote.
```

Comment:
Not sure what exactly you mean by skip, but even without differences in remote chains, this won't be contiguous if there are multiple messages in the same block (range)

Reply:
Okay, maybe we need a better word but contiguous in that each +1 to the index represents 1 messages. +5 still means 5 more were sent from the origin, but +5 to the remote could indicate only 1 was sent to that remote. It is just the index of the last message basically.